backend/inspirehep/records/marshmallow/seminars/ui.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+from inspirehep.accounts.api import is_superuser_or_cataloger_or_submitter_logged_in
+from inspirehep.records.marshmallow.seminars.base import SeminarsPublicSchema
+
+
+class SeminarsBaseSchema(SeminarsPublicSchema):
+    def __init__(self, *args, **kwargs):
+        super().__init__()
+        self.post_dumps.append(self.set_can_edit)
+
+    @staticmethod
+    def set_can_edit(data, original_data):
+        if is_superuser_or_cataloger_or_submitter_logged_in(data):
```

Comment:
also the name of this function can be just `can_user_edit_record()`

Reply:
I named it like this because it doesn't seem unlikely to me that we would use this function in other contexts as well, not only ```can_edit``` but I agree the name is a bit long

Reply:
it seems unlikely to me? and passing whole record to a function named `is_superuser_or_cataloger_or_submitter_logged_in` doesn't seem very straightforward